### PR TITLE
feat: propose OR where cond query.

### PR DESCRIPTION
**Pull request by Colstuwjx**
This PR is going to propose an OR operator implementation.

### CODE REVIEW

**Comment by Colstuwjx** [comment]
It's still in progress. After I'm very sure to merge, I'll mark it ready for review.

File: builder/builder.go [hunk 1]
```diff
@@ -220,9 +221,32 @@ func getWhereConditions(where map[string]interface{}) ([]Comparable, func(), err
...
+			)
+			if orWheres, ok = val.([]map[string]interface{}); !ok {
+				return nil, emptyFunc, errOrValueType
+			}
+			for _, orWhere := range orWheres {
+				if orWhere == nil {
+					continue
+				}
+				orNestWhere, f, err := getWhereConditions(orWhere)
+				defer f() // ?
```
Comment (Colstuwjx) [inline builder/builder.go:242]: We're using a global variable `cpPool`, so I wonder if this line may doesn't work.

**Comment by Colstuwjx** [comment]
@caibirdme The testcase seems passed, could you help me review this PR?
Thanks.

**Comment by caibirdme** [comment]
> @caibirdme The testcase seems passed, could you help me review this PR?
> Thanks.

I really appreciate for your contribution. I'll review this pr later this afternoon, and I hope you can add more test cases as we did in [test](https://github.com/didi/gendry/blob/master/builder/builder_test.go#L273). Make the test cases more intuitive

**Comment by Colstuwjx** [comment]
Thank you for pointing out, it seems the `Test_BuildSelect` would broken if I add `_or` statements, I'll fix it soon.

**Comment by Colstuwjx** [comment]
Fixed issue and squashed commits!

**Comment by caibirdme** [comment]
I created a new branch `feature-remove-pool`, and in this branch, I delete the sync.Pool. Could you pull the branch and see if you can pass the tests

**Comment by Colstuwjx** [comment]
> I created a new branch feature-remove-pool, and in this branch, I delete the sync.Pool. Could you pull the branch and see if you can pass the tests

I'm not sure it's a good idea that introduce two changes(_or operator support and remove sync.pool) inside one PR. If it's OK, I'll try cherry-pick your commit and test it.

**Comment by caibirdme** [comment]
> > I created a new branch feature-remove-pool, and in this branch, I delete the sync.Pool. Could you pull the branch and see if you can pass the tests
> 
> I'm not sure it's a good idea that introduce two changes(_or operator support and remove sync.pool) inside one PR. If it's OK, I'll try cherry-pick your commit and test it.

Yep, it's not a good idea. I mean, you can test your pr on that branch, and I'll try to test and merge that branch at the same time. After merging that branch into master, it's also easy to merge your pr based on that branch

**Comment by Colstuwjx** [comment]
@caibirdme I've been created a new PR #78  to demonstrate these two changes, it seems works well!

File: builder/builder.go (hunk 1 above)
Comment (Colstuwjx) [reply builder/builder.go:242]: I've been ran the golang race detector, and it seems no race detected (without your prompted changes):

```
➜  builder git:(build-select-or-operator) go test -race *.go

ok  	command-line-arguments	1.984s
```

I understand with your concerns, and I also agree that merge your `remove-pool` PR first, and then merge this PR after rebased master.

Thanks!

**Comment by Colstuwjx** [comment]
@caibirdme  Fixed all issues, and wish you could help me check again.

**Comment by caibirdme** [comment]
Merged, thx for your contribution!!! And I will create a new tag one or two weeks later if there's no bug report

**Pull request merged**